terraso_backend/apps/auth/views.py

```diff
@@ -114,18 +128,33 @@ def process_callback(self, req):
             logger.exception("Error attempting create access and refresh tokens")
             return HttpResponse(f"Error: {exc}", status=400)
 
-        redirect_uri = f"{settings.WEB_CLIENT_URL}/{self.state}"
+        origin = self.state["origin"]
+        redirect_uri = self.state["redirectUrl"]
         if cookie := req.get_signed_cookie(
             OAUTH_COOKIE_NAME, None, max_age=OAUTH_COOKIE_MAX_AGE_SECONDS
         ):
             # we stored the original URI as a cookie
             # redirecting the user to this URI will start our OAuth process
             if self._check_cookie(cookie):
                 redirect_uri = cookie
-        response = HttpResponseRedirect(redirect_uri)
+
+        # Get the domain from the redirect URI or use configured domain
+        redirect_domain = urlparse(origin).hostname
+        if settings.ENV == "production" or redirect_domain == settings.WEB_CLIENT_DOMAIN:
```

Comment:
Question: Can we make production and staging use the regular more secure method? and just use this for the preview branches? I think my concern is not having the production auth flow in another environment, and maybe this makes sense for staginig since theoretically staging should be as close as possible to production

Reply:
I think this is enoguh for me for this:

> the staging frontend instance will cause the backend to follow the same path is prod

So, if that is the case I think we are having that same behavior on staging, still not a strong opinion, just a nice to have